src/seedu/addressbook/data/person/Address.java

```diff
@@ -44,28 +48,22 @@ public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
+    /**
+     * @return String representation of address
+     */
     @Override
     public String toString() {
-        String[] addressParts = new String[] { block.toString(), street.toString(), unit.toString(), postalCode.toString() };
-        String ret = "";
-        for(int i = 0; i < addressParts.length; i++){
-            String part = addressParts[i];
-            if(!part.equals("")){
-                if(!ret.equals("")){
-                    ret += ", ";
-                }
-                ret += part;
-            };
-        }
-        return ret;
+        String[] addressParts = new String[] { block.toString(), street.toString(), unit.toString(),
+                postalCode.toString() };
+        return String.join(", ", addressParts);
     }
-
+    
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
                         && this.toString().equals(((Address) other).toString())); // state
-                                                                                  // check
+        // check
```

Comment:
this is due to the automatic formatting in Eclipse, will disable comment formatting